examples/CantileverDistributedLoad/cantilever_distrubuted_load_postprecessing.py

```diff
@@ -0,0 +1,396 @@
+import numpy as np
+from matplotlib import pyplot as plt
+from matplotlib import cm
+from tqdm import tqdm
+from typing import Dict, Sequence
+from typing import Optional
+import logging
+from numpy.testing import assert_allclose
+from elastica.utils import MaxDimension, Tolerance
+from elastica._linalg import _batch_cross, _batch_norm, _batch_dot
+from elastica.rod.factory_function import (
+    _assert_dim,
+    _position_validity_checker,
+    _directors_validity_checker,
+    _position_validity_checker_ring_rod,
+)
+
+
+def Find_Tip_Position(rod, n_elem):
+    x_tip = rod.position_collection[0][n_elem]
+    y_tip = rod.position_collection[1][n_elem]
+    z_tip = rod.position_collection[2][n_elem]
+
+    return x_tip, y_tip, z_tip
+
+
+def adjust_square_cross_section(
```

Comment:
I had tried lots of ways to simplify this process, but I found it hardly possible. Because I need to change all three second moment of mass (I0_1,I0_2,I0_3) according to the shape of the cross-section, and all other key parameters including bend_matrix and shear_matrix, needed to be recalculated based on the new second moment of mass. So it's basically the same as the factory_function. However, I do can take the rod as an argument.

Reply:
That's fine. A few things you can do:

1. Like you said, take the rod as an argument, and the function need not return anything since rod is passed in as reference
2. Many of the initial parts of this function is already done in `factory_function`, which is called when you first initialize the rods. Since you are only modifying moi, bend and shear matrices, there is no need for things like position/director initializations and all those assertions/validity checks as they are readily done in `factory_function`. Only keep the most essential parts in this function. 
3. You might consider merging the two cantilever cases into one directory. The post-processing files of these two are almost identical as far as i can tell. 
4. I see you hard-coded `side_length` and `poisson_ratio` into this function. Consider passing them in as arguments so the function is generic and not tied to this particular case